bigquery/google/cloud/bigquery/job.py

```diff
@@ -542,35 +540,153 @@ def cancelled(self):
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.load.encoding
+    """
+
+    field_delimiter = _TypedApiResourceProperty(
+        'field_delimiter', 'fieldDelimiter', six.string_types)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.load.fieldDelimiter
+    """
+
+    ignore_unknown_values = _TypedApiResourceProperty(
+        'ignore_unknown_values', 'ignoreUnknownValues', bool)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.load.ignoreUnknownValues
+    """
+
+    max_bad_records = _TypedApiResourceProperty(
+        'max_bad_records', 'maxBadRecords', six.integer_types)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.load.maxBadRecords
+    """
+
+    null_marker = _TypedApiResourceProperty(
+        'null_marker', 'nullMarker', six.string_types)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.load.nullMarker
+    """
+
+    quote_character = _TypedApiResourceProperty(
+        'quote_character', 'quote', six.string_types)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.load.quote
+    """
+
+    skip_leading_rows = _TypedApiResourceProperty(
+        'skip_leading_rows', 'skipLeadingRows', six.integer_types)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.load.skipLeadingRows
+    """
+
+    source_format = SourceFormat('source_format', 'sourceFormat')
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.load.sourceFormat
+    """
+
+    write_disposition = WriteDisposition('write_disposition',
+                                         'writeDisposition')
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.load.writeDisposition
+    """
+
+    @property
+    def schema(self):
+        """See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.load.schema
+        """
+        return list(self._schema)
+
+    @schema.setter
+    def schema(self, value):
+        if not all(isinstance(field, SchemaField) for field in value):
+            raise ValueError('Schema items must be fields')
+        if self.autodetect:
+            raise ValueError(
+                'Schema can not be set if `autodetect` property is True')
+        self._schema = tuple(value)
+
+    def to_api_repr(self):
+        """Build an API representation of the load job config.
+
+        :rtype: dict
+        :returns: A dictionary in the format used by the BigQuery API.
+        """
+        config = copy.deepcopy(self._properties)
+        if len(self.schema) > 0:
+            config['schema'] = {'fields': _build_schema_resource(self.schema)}
+        # TODO(jba): add a comment explaining why skipLeadingRows is a string.
```

Comment:
I bet the reason is that the proto file in the backend accepts int64, which isn't representable as a JSON number.

Reply:
Ack.

Reply:
Indeed: https://developers.google.com/discovery/v1/type-format